### DISCUSSION TITLE
How to insert <TAB> after word?

### DISCUSSION THREAD

**Discussion by funny-falcon** [helix-editor/helix#12133] [General] [answered]
Our coding standard require to insert <TAB> after type name on variable/field declaration:
```c
struct A {
  size_t/*TAB here*/fsize;
  size_t/*and I try to type it here*/
};
```
When I type `size_t` autocompletion window pops up, and <TAB> starts choose of variants, and I don't get how to insert just <TAB> itself.

Can you help me?

**Comment by kirawi** [comment, marked as answer]
Do `C-c` to close the popup and then try to insert with `S-tab`. (By default, pressing tab when not at the whitespace beginning of a line will use smart-tab behavior where the cursor will jump to the end of the current syntax node. But this can be overridden as shown in https://docs.helix-editor.com/editor.html#editorsmart-tab-section)

**Reply by funny-falcon** [reply]
Yes, looks like.

Still is it possible to completely unmap Tab from variant choosing (without enabling smart-tab, which I found very confusing as well, so disabled it)?

**Reply by kirawi** [reply]
Hmm. You can try to reenable smart-tab, enable `editor.smart-tab.supersede-menu` and bind `tab = "insert_tab"`. A PR to make supersede-menu work without having smart-tab enabled might be a good idea too though.
